central/imageintegration/datastore/singleton.go

```diff
@@ -75,6 +78,65 @@ func deleteAutogeneratedRegistries(ctx context.Context, iiStore store.Store, iis
 	log.Info("Completed deletion of image integrations")
 }
 
+// setupScannerV4Integration will ensure the default Scanner V4 integration exists if
+// Scanner V4 is installed/enabled (forces opt-in).  If Scanner V4 is not installed (or is disabled)
+// will delete all ScannerV4 integrations.
+func setupScannerV4Integration(ctx context.Context, iiStore store.Store, iis []*storage.ImageIntegration) {
+	keepDefault := true // for readability.
+
+	// If Scanner V4 is not installed delete the associated integration(s).
+	if !features.ScannerV4Enabled.Enabled() {
+		deleteScannerV4Integrations(ctx, iiStore, iis, !keepDefault)
+		return
+	}
+
+	// Create the default Scanner V4 integration if it doesn't exist.
+	createScannerV4Integration(ctx, iiStore)
+
+	// Delete all but the default Scanner V4 integration (should be none).
+	deleteScannerV4Integrations(ctx, iiStore, iis, keepDefault)
+}
+
+// createScannerV4Integration will create the default Scanner V4 integration if it does
+// not currently exist.
+func createScannerV4Integration(ctx context.Context, iiStore store.Store) {
+	if _, exists, err := iiStore.Get(ctx, store.DefaultScannerV4Integration.GetId()); err != nil {
+		log.Errorf("Unable to detect if default Scanner V4 integration exists: %v", err)
+		return
+	} else if exists {
+		// Nothing to do, integration exists.
+		return
+	}
+
+	log.Infof("Upserting default Scanner V4 integration %q (%v)", store.DefaultScannerV4Integration.GetName(), store.DefaultScannerV4Integration.GetId())
+	if err := iiStore.Upsert(ctx, store.DefaultScannerV4Integration); err != nil {
+		log.Errorf("Unable to upsert default Scanner V4 integration: %v", err)
+	}
+}
+
+// deleteScannerV4Integrations will delete all Scanner V4 integrations except for the
+// the default integration if keepDefault is true.
+func deleteScannerV4Integrations(ctx context.Context, iiStore store.Store, iis []*storage.ImageIntegration, keepDefault bool) {
```

Comment:
is this for future-proofing in case we change the default?

Reply:
This is to address a scenario where somehow an extra v4 integration was created (perhaps by inserting into the DB bypassing the API).

The logic for pulling vuln status, cert status, etc. assumes there is only one scanner of type `clairify` today, with `scannerv4` we're only allowing a single `scannerv4` to be defined to avoid ambiguity (more details on this in the design).